### ISSUE TITLE
tqdm doesn't print dictionary values when setting self.automatic_optimization as False

### ISSUE THREAD

**Issue by sainatarajan** [PyTorchLightning/pytorch-lightning#11693]
## 🐛 Bug

The tqdm progress bar doesn't display values such as losses/metrics when `self.automatic_optimization= False`.

### To Reproduce

```python
import os
os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'

import pytorch_lightning as pl
import torch
from pytorch_lightning import LightningModule
from pytorch_lightning.utilities.seed import seed_everything
from torch.utils.data import DataLoader, Dataset

from collections import OrderedDict


class RandomDataset(Dataset):
    def __init__(self, size, num_samples):
        self.len = num_samples
        self.x = torch.randn(num_samples, size)
        self.y = self.x[:, 0] > .5

    def __getitem__(self, index):
        return self.x[index], self.y[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):

    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 1)
        self.automatic_optimization = False

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        opt = self.optimizers()
        opt.zero_grad()
        x, y = batch
        output = self.layer(x)
        loss = (output - y.to(torch.float32)).square().mean()

        self.manual_backward(loss)
        opt.step()
        tqdm_dict = {'train_loss': loss.item()}
        output = {"loss": loss, "progress_bar": tqdm_dict, "log": tqdm_dict}

        return output

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        return optimizer


seed_everything(seed=0)

# init model
model = BoringModel()

# Initialize a trainer
trainer = pl.Trainer(
    max_epochs=200,
    gpus=1,
)

train = RandomDataset(32, 1000)
train = DataLoader(train, batch_size=16)

# Train the model
trainer.fit(model, train)
```

### Expected behavior

Epoch 24:  38%|██████████████████████████████████████████████████████████████▊                                                                                                      | 24/63 [00:00<00:00, 186.50it/s, loss=0.223, v_num=8]

### Environment
```
* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 2060
        - available:         True
        - version:           11.3
* Packages:
        - numpy:             1.21.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.1+cu113
        - pytorch-lightning: 1.5.9
        - tqdm:              4.62.3
* System:
        - OS:                Windows
        - architecture:
                - 64bit
                - WindowsPE
        - processor:         Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
        - python:            3.8.5
        - version:           10.0.19041
```

### Additional context

cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy @awaelchli

**Comment by ananthsub** [comment]
If you use `self.log` instead of returning structured outputs from the training step, do you see the values in the progress bar?
```py
    def training_step(self, batch, batch_idx):
        opt = self.optimizers()
        opt.zero_grad()
        x, y = batch
        output = self.layer(x)
        loss = (output - y.to(torch.float32)).square().mean()

        self.manual_backward(loss)
        opt.step()
        self.log("train_loss", loss, prog_bar=True)
        return loss
```

**Comment by carmocca** [comment]
@sainatarajan the structured outputs you are returning are for an older (legacy) Lightning version.

The snippet shared in the comment above should work without issues.

Would you mind sharing here where you found this implementation?

**Comment by sainatarajan** [comment]
@ananthsub Your script seems to work! Thanks.
@carmocca Thanks for your comment. I'm not sure about it. I think I found it [here](https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/basic-gan.html).

**Comment by rohitgr7** [comment]
we need to review and fix those tutorials with the latest changes.
